dipy/segment/tests/test_mrf.py

```diff
@@ -0,0 +1,371 @@
+import numpy as np
+import numpy.testing as npt
+from dipy.data import get_data
+from dipy.sims.voxel import add_noise
+from dipy.segment.mrf import (ConstantObservationModel,
+                              IteratedConditionalModes)
+from dipy.segment.tissue import (TissueClassifierHMRF)
+
+
+# Load a coronal slice from a T1-weighted MRI
+fname = get_data('t1_coronal_slice')
+single_slice = np.load(fname)
+
+# Stack a few copies to form a 3D volume
+nslices = 5
+image = np.zeros(shape=single_slice.shape + (nslices,))
+image[..., :nslices] = single_slice[..., None]
+
+# Set up parameters
+nclasses = 4
+beta = np.float64(0.0)
+max_iter = 10
+background_noise = True
+
+# Making squares
+square = np.zeros((256, 256, 3))
+square[42:213, 42:213, :] = 1
+square[71:185, 71:185, :] = 2
+square[99:157, 99:157, :] = 3
+
+square_gauss = np.zeros((256, 256, 3)) + 0.001
+square_gauss = add_noise(square_gauss, 10000, 1, noise_type='gaussian')
+square_gauss[42:213, 42:213, :] = 1
+noise_1 = np.random.normal(1.001, 0.0001,
+                           size=square_gauss[42:213, 42:213, :].shape)
+square_gauss[42:213, 42:213, :] = square_gauss[42:213, 42:213, :] + noise_1
+square_gauss[71:185, 71:185, :] = 2
+noise_2 = np.random.normal(2.001, 0.0001,
+                           size=square_gauss[71:185, 71:185, :].shape)
+square_gauss[71:185, 71:185, :] = square_gauss[71:185, 71:185, :] + noise_2
+square_gauss[99:157, 99:157, :] = 3
+noise_3 = np.random.normal(3.001, 0.0001,
+                           size=square_gauss[99:157, 99:157, :].shape)
+square_gauss[99:157, 99:157, :] = square_gauss[99:157, 99:157, :] + noise_3
+
+square_1 = np.zeros((256, 256, 3)) + 0.001
+square_1 = add_noise(square_1, 10000, 1, noise_type='gaussian')
+temp_1 = np.random.random_integers(20, size=(171, 171, 3))
+temp_1 = np.where(temp_1 < 20, 1, 3)
+square_1[42:213, 42:213, :] = temp_1
+temp_2 = np.random.random_integers(20, size=(114, 114, 3))
+temp_2 = np.where(temp_2 < 19, 2, 1)
+square_1[71:185, 71:185, :] = temp_2
+temp_3 = np.random.random_integers(20, size=(58, 58, 3))
+temp_3 = np.where(temp_3 < 20, 3, 1)
+square_1[99:157, 99:157, :] = temp_3
+
+
+def test_greyscale_image():
+
+    com = ConstantObservationModel()
+    icm = IteratedConditionalModes()
+
+    mu, sigma = com.initialize_param_uniform(image, nclasses)
+    sigmasq = sigma ** 2
+    npt.assert_equal(mu, np.array([0., 0.25, 0.5, 0.75]))
```

Comment:
In general, we should not compare floating point values for equality, we should use `assert_array_almost_equal` instead.